Doc/whatsnew/3.14.rst

```diff
@@ -1094,135 +1039,173 @@ Platform support
 
-* Three-argument :func:`pow` now tries calling :meth:`~object.__rpow__` if
-  necessary. Previously it was only called in two-argument :func:`!pow` and the
-  binary power operator.
+* Three-argument :func:`pow` now tries calling :meth:`~object.__rpow__`
+  if necessary.
+  Previously it was only called in two-argument :func:`!pow`
+  and the binary power operator.
   (Contributed by Serhiy Storchaka in :gh:`130104`.)
 
-* Add a built-in implementation for HMAC (:rfc:`2104`) using formally verified
-  code from the `HACL* <https://github.com/hacl-star/hacl-star/>`__ project.
-  This implementation is used as a fallback when the OpenSSL implementation
-  of HMAC is not available.
-  (Contributed by Bénédikt Tran in :gh:`99108`.)
+* :class:`super` objects are now :mod:`copyable <copy>` and :mod:`pickleable
+  <pickle>`.
+  (Contributed by Serhiy Storchaka in :gh:`125767`.)
+
+
+Command line and environment
+----------------------------
 
 * The import time flag can now track modules that are already loaded ('cached'),
   via the new :option:`-X importtime=2 <-X>`.
   When such a module is imported, the ``self`` and ``cumulative`` times
   are replaced by the string ``cached``.
+
   Values above ``2`` for ``-X importtime`` are now reserved for future use.
-  (Contributed by Noah Kim and Adam Turner in :gh:`118655`.)
 
-* When subclassing from a pure C type, the C slots for the new type are no
-  longer replaced with a wrapped version on class creation if they are not
-  explicitly overridden in the subclass.
-  (Contributed by Tomasz Pytel in :gh:`132329`.)
+  (Contributed by Noah Kim and Adam Turner in :gh:`118655`.)
 
 * The command-line option :option:`-c` now automatically dedents its code
   argument before execution. The auto-dedentation behavior mirrors
   :func:`textwrap.dedent`.
   (Contributed by Jon Crall and Steven Sun in :gh:`103998`.)
 
-* Improve error message when an object supporting the synchronous
-  context manager protocol is entered using :keyword:`async
-  with` instead of :keyword:`with`.
-  And vice versa with the asynchronous context manager protocol.
-  (Contributed by Bénédikt Tran in :gh:`128398`.)
-
 * :option:`!-J` is no longer a reserved flag for Jython_,
   and now has no special meaning.
   (Contributed by Adam Turner in :gh:`133336`.)
 
   .. _Jython: https://www.jython.org/
 
-* The :func:`int` built-in no longer delegates to :meth:`~object.__trunc__`.
-  Classes that want to support conversion to :func:`!int` must implement
-  either :meth:`~object.__int__` or :meth:`~object.__index__`.
-  (Contributed by Mark Dickinson in :gh:`119743`.)
 
-* Using :data:`NotImplemented` in a boolean context
-  will now raise a :exc:`TypeError`.
-  This has raised a :exc:`DeprecationWarning` since Python 3.9.
-  (Contributed by Jelle Zijlstra in :gh:`118767`.)
+.. _whatsnew314-finally-syntaxwarning:
 
+Control flow in :keyword:`finally` blocks
+-----------------------------------------
```

Comment:
I think it's useful to keep the PEP number in the title:
```suggestion
PEP 765: Control flow in :keyword:`finally` blocks
--------------------------------------------------
```

Reply:
I deliberately removed it, it's not relevant to the reader unless they want further detail, at which point you have the link within the body.

I also think emphasising the PEP number can downplay the import of other big changes/features that didn't have a PEP. Again this is the first of these changes.

A

Reply:
In my mind, this is because a PEP does not necessarily denote importance to the end user, but that we as the core developers used it as a useful decision making document. For example, many of the C API changes have PEPs, but are not relevant to 99%+ of Python end users, who do not write C code. Further, improved error messages have been a lauded feature of the past few releases, yet have never had a PEP, nor did any of the REPL changes, nor colourisation, nor remote debugging, nor the various asyncio changes in 3.14. I could go on! 

Perhaps some of these items ought to have had PEPs, but I don't want to diminish their importance by a somewhat arbitrary PEP/no-PEP divide, which putting the PEP number in the subhead would do.

Reply:
I still believe we shoud make that distinction. Note that the release highlights contain the PEP numbers. Not finding them in the sections list would be less helpful for readers. Please keep them.

Reply:
To unblock this PR, I will add the PEP number back here, but I still think it's the wrong choice, for the various reasons enumerated above. I'll propose a single PR removing the numbers everywhere later on, so that it can be decided in a single up/down change.